packages/loot-core/src/server/accounts/transaction-rules.ts

```diff
@@ -291,6 +291,24 @@ export async function runRules(trans) {
   return await finalizeTransactionForRules(finalTrans);
 }
 
+function conditionSpecialCases(cond: Condition): Condition[] {
+  //special cases that require multiple conditions
+  if (cond.op === 'is' && cond.field === 'category' && cond.value === null) {
+    return [
+      cond,
+      new Condition('is', 'transfer', false, null),
+      new Condition('is', 'parent', false, null),
+    ];
```

Comment:
I don't think this will work when a rules uses `If any of these conditions match:`

Reply:
You are correct. I will rework this then.

Reply:
This has been updated to allow a new condition operator of subExpression, so the grouped conditions will be properly contained in an and when passed to the rule engine. This still preserves the desired functionality of the account filtering.